codegen/src/main/scala/akka/grpc/gen/CodeGenerator.scala

```diff
@@ -23,3 +28,28 @@ trait CodeGenerator {
   final def run(request: Array[Byte], logger: Logger): Array[Byte] =
     run(CodeGeneratorRequest.parseFrom(request), logger: Logger).toByteArray
 }
+
+private[gen] object CodeGenerator {
+
+  /** Extract the longest common package prefix for a list of packages. */
+  private[gen] def commonPackage(packages: Seq[String]): String =
+    packages.reduce(commonPackage(_, _))
```

Comment:
The old code had an optimisation to check everything for equality. I've moved that optimisation out of this method and into the `commonPackage(String,String)` method. That means we can use `reduce` instead of writing our own fold.